### PR TITLE
[4092] Expose MessageOptionsDialogFragment

### LINKED ISSUE #4092: Consider exposing MessageOptionsDialogFragment
### Context & proposal

Check that it's possible and reasonable to make `MessageOptionsDialogFragment` public.

### Acceptance criteria

To accept this task we need to complete the following items:

- [x] Clients are able to use MessageOptionsDialogFragment

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

Closes #4092

Expose `MessageOptionsDialogFragment` so that clients are able to lunch the fragment manually.

### 🧪 Testing

1. Apply one of the patches below
2. Long press on a message to show the dialog
3. Click on the message option

<details>

<summary>Patch file to apply these changes for testing</summary>

```
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(revision 70613ef09110502269b3ddde34cc95472591ab6a)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(date 1662062195662)
@@ -37,6 +37,7 @@
 import io.getstream.chat.android.common.model.DeleteMessage
 import io.getstream.chat.android.common.model.EditMessage
 import io.getstream.chat.android.common.model.SendAnyway
+import io.getstream.chat.android.common.state.CustomAction
 import io.getstream.chat.android.common.state.DeletedMessageVisibility
 import io.getstream.chat.android.common.state.Edit
 import io.getstream.chat.android.common.state.MessageMode
@@ -49,9 +50,12 @@
 import io.getstream.chat.android.ui.message.input.viewmodel.bindView
 import io.getstream.chat.android.ui.message.list.header.viewmodel.MessageListHeaderViewModel
 import io.getstream.chat.android.ui.message.list.header.viewmodel.bindView
+import io.getstream.chat.android.ui.message.list.options.message.MessageOptionItem
 import io.getstream.chat.android.ui.message.list.viewmodel.bindView
 import io.getstream.chat.android.ui.message.list.viewmodel.factory.MessageListViewModelFactory
+import io.getstream.chat.ui.sample.R
 import io.getstream.chat.ui.sample.common.navigateSafely
+import io.getstream.chat.ui.sample.common.showToast
 import io.getstream.chat.ui.sample.databinding.FragmentChatBinding
 import io.getstream.chat.ui.sample.feature.common.ConfirmationDialogFragment
 import io.getstream.chat.ui.sample.util.extensions.useAdjustResize
@@ -276,6 +280,27 @@
                 }
             }
         }
+
+
+        binding.messageListView.setMessageLongClickListener { message ->
+            binding.messageListView.showMessageOptionsDialog(
+                childFragmentManager,
+                message,
+                messageOptionItems = listOf(
+                    MessageOptionItem(
+                        optionText = "Custom option",
+                        optionIcon = requireContext().getDrawable(R.drawable.stream_ui_ic_command)!!,
+                        messageAction = CustomAction(
+                            message = message,
+                            extraProperties = mapOf("key1" to "value1")
+                        ),
+                    )
+                ),
+                optionClickListener = {
+                    showToast("Custom Action!")
+                }
+            )
+        }
     }
 
     private fun shouldShowDateSeparator(calendar: Calendar, previousMessage: Message, message: Message): Boolean {

```

</details>

<details>
  <summary>Patch file to apply these changes for testing</summary>

```
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(revision 70613ef09110502269b3ddde34cc95472591ab6a)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt	(date 1662062652639)
@@ -37,6 +37,7 @@
 import io.getstream.chat.android.common.model.DeleteMessage
 import io.getstream.chat.android.common.model.EditMessage
 import io.getstream.chat.android.common.model.SendAnyway
+import io.getstream.chat.android.common.state.CustomAction
 import io.getstream.chat.android.common.state.DeletedMessageVisibility
 import io.getstream.chat.android.common.state.Edit
 import io.getstream.chat.android.common.state.MessageMode
@@ -49,9 +50,13 @@
 import io.getstream.chat.android.ui.message.input.viewmodel.bindView
 import io.getstream.chat.android.ui.message.list.header.viewmodel.MessageListHeaderViewModel
 import io.getstream.chat.android.ui.message.list.header.viewmodel.bindView
+import io.getstream.chat.android.ui.message.list.options.message.MessageOptionItem
+import io.getstream.chat.android.ui.message.list.options.message.internal.MessageOptionsDialogFragment
 import io.getstream.chat.android.ui.message.list.viewmodel.bindView
 import io.getstream.chat.android.ui.message.list.viewmodel.factory.MessageListViewModelFactory
+import io.getstream.chat.ui.sample.R
 import io.getstream.chat.ui.sample.common.navigateSafely
+import io.getstream.chat.ui.sample.common.showToast
 import io.getstream.chat.ui.sample.databinding.FragmentChatBinding
 import io.getstream.chat.ui.sample.feature.common.ConfirmationDialogFragment
 import io.getstream.chat.ui.sample.util.extensions.useAdjustResize
@@ -276,6 +281,34 @@
                 }
             }
         }
+
+        binding.messageListView.setMessageLongClickListener { message ->
+            MessageOptionsDialogFragment.newInstance(
+                context = requireContext(),
+                message = message,
+                messageOptionItems = listOf(
+                    MessageOptionItem(
+                        optionText = "Custom option",
+                        optionIcon = requireContext().getDrawable(R.drawable.stream_ui_ic_command)!!,
+                        messageAction = CustomAction(
+                            message = message,
+                            extraProperties = mapOf("key1" to "value1")
+                        ),
+                    )
+                ),
+            ).apply {
+                setReactionClickListener { message, reactionType ->
+                    messageListViewModel.onEvent(
+                        MessageListViewModel.Event.MessageReaction(message, reactionType, true)
+                    )
+                }
+                setMessageOptionClickListener { messageAction ->
+                    if (messageAction is CustomAction) {
+                        showToast("Custom Action!")
+                    }
+                }
+            }.show(childFragmentManager, MessageOptionsDialogFragment.TAG)
+        }
     }
 
     private fun shouldShowDateSeparator(calendar: Calendar, previousMessage: Message, message: Message): Boolean {

```

</details>

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/internal/FullScreenDialogFragment.kt [hunk 1]
```diff
@@ -27,7 +27,7 @@ import androidx.fragment.app.DialogFragment
 /**
  * A fragment that displays a full screen overlay dialog.
  */
-internal open class FullScreenDialogFragment : DialogFragment() {
+public open class FullScreenDialogFragment : DialogFragment() {
```
Comment (skydoves) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/internal/FullScreenDialogFragment.kt:30]: Now can we move this class out of the `internal` package?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/options/message/internal/MessageOptionsDialogFragment.kt [hunk 2]
```diff
@@ -33,37 +34,44 @@ import io.getstream.chat.android.client.models.Message
...
 import io.getstream.chat.android.ui.message.list.background.MessageBackgroundFactory
+import io.getstream.chat.android.ui.message.list.background.MessageBackgroundFactoryImpl
 import io.getstream.chat.android.ui.message.list.options.message.MessageOptionItem
 
 /**
  * An overlay with available message options to the selected message. Also, allows leaving a reaction.
  */
-internal class MessageOptionsDialogFragment : FullScreenDialogFragment() {
+@ExperimentalStreamChatApi
+public class MessageOptionsDialogFragment : FullScreenDialogFragment() {
```
Comment (skydoves) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/options/message/internal/MessageOptionsDialogFragment.kt:66]: Same here do we need to move this out of the `internal` package?

**Review by skydoves — COMMENTED** [review]
Overall looks good to me! I left some trivial comments.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/options/message/internal/MessageOptionsDialogFragment.kt (hunk 2 above)
Comment (filbabic) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/options/message/internal/MessageOptionsDialogFragment.kt:66]: sounds good

**Review by filbabic — APPROVED** [review]
- [x] UI Components sample runs & works
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**